### PR TITLE
[#2755] Authenticate controller

### LINKED ISSUE #2755: Authentication to the Airy controller
**Background:** [See milestone](https://github.com/airyhq/airy/milestone/36)

**Motivation:** Since we need to expose controller APIs to clients we need to be able to authenticate against it.

- [x] Support Airy authentication (system token and jwt) on the controller service
- [x] Enable internal requests from `api-admin` to still work
- [ ] Add an `airy login` command so that the CLI can obtain system token authentication 
 

Perhaps we can use this routing lib: https://github.com/gorilla/mux

### PR DESCRIPTION
**Pull request by chrismatix**
required for #2755 

The `airy login` command will follow in a separate PR

### CODE REVIEW

File: infrastructure/controller/pkg/endpoints/server.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+	"k8s.io/client-go/kubernetes"
+)
+
+func Serve(clientSet *kubernetes.Clientset, namespace string) {
+	r := mux.NewRouter()
+
+	// Load authentication middleware only if auth env is present
+	authEnabled := false
+	systemToken := os.Getenv("systemToken")
```
Comment (ljupcovangelski) [inline infrastructure/controller/pkg/endpoints/server.go:18]: Should we put this ENV also in the deployment.yaml file, from the security configMap?

**Review by ljupcovangelski — COMMENTED** [review]
Looks great :+1:

File: infrastructure/controller/pkg/endpoints/server.go (hunk 1 above)
Comment (chrismatix) [reply infrastructure/controller/pkg/endpoints/server.go:18]: Good catch

**Review by ljupcovangelski — DISMISSED** [review]

**Review by juan-sebastian — COMMENTED** [review]
Good overall. Left some minor comments

**Review by juan-sebastian — APPROVED** [review]

**Review by ljupcovangelski — APPROVED** [review]

**Pull request merged**